controllers/verticalscalingpolicy_controller.go

```diff
@@ -136,54 +141,86 @@ func (r *VerticalScalingPolicyReconciler) Reconcile(ctxt context.Context, req ct
 		currentCapacityUtilization := 100 * (totalCPUCapacityFloat - totalCPUAllocatableFloat) / totalCPUCapacityFloat
 		currentMemoryUtilization := 100 * (totalMemoryCapacityFloat - totalMemoryAllocatableFloat) / totalMemoryCapacityFloat
 
-		scaleUpBehaviorPolicies := vsp.Spec.Behavior.ScaleUp.Policies // TODO: Make sure to avoid null pointer exception here by validating vsp before this
-		scaleUpOnNodesCountPolicy := getBehaviorPolicy(scaleUpBehaviorPolicies, v1alpha1.NodesCountUtilizationPercent)
-		scaleUpOnCpuUtilization := getBehaviorPolicy(scaleUpBehaviorPolicies, v1alpha1.CPUUtilizationPercent)
-		scaleUpOnMemoryUtilization := getBehaviorPolicy(scaleUpBehaviorPolicies, v1alpha1.MemoryUtilizationPercent)
+		scaleUpOnNodesCountPolicy := vsp.Spec.Behavior.ScaleUp.GetPolicy(v1alpha1.NodesCountUtilizationPercent)
+		scaleUpOnCpuUtilizationPolicy := vsp.Spec.Behavior.ScaleUp.GetPolicy(v1alpha1.CPUUtilizationPercent)
+		scaleUpOnMemoryUtilizationPolicy := vsp.Spec.Behavior.ScaleUp.GetPolicy(v1alpha1.MemoryUtilizationPercent)
 
-		scaleDownBehaviorPolicies := vsp.Spec.Behavior.ScaleDown.Policies // TODO: Make sure to avoid null pointer exception here by validating vsp before this
-		scaleDownOnCpuUtilization := getBehaviorPolicy(scaleDownBehaviorPolicies, v1alpha1.CPUUtilizationPercent)
-		scaleDownOnMemoryUtilization := getBehaviorPolicy(scaleDownBehaviorPolicies, v1alpha1.MemoryUtilizationPercent)
+		scaleDownOnCpuUtilizationPolicy := vsp.Spec.Behavior.ScaleDown.GetPolicy(v1alpha1.CPUUtilizationPercent)
+		scaleDownOnMemoryUtilizationPolicy := vsp.Spec.Behavior.ScaleDown.GetPolicy(v1alpha1.MemoryUtilizationPercent)
 
 		scaleUp_stabilizationWindow := time.Duration(vsp.Spec.Behavior.ScaleUp.StabilizationWindowSeconds) * time.Second
 		scaleDown_stabilizationWindow := time.Duration(vsp.Spec.Behavior.ScaleDown.StabilizationWindowSeconds) * time.Second
 
 		// If there is a larger instance type available, check if we want to vertically scale up the IG
-		if hasLargerInstanceType && vsp.Status != nil && time.Since(vsp.Status.TargetStatuses[igName].LastTransitionTime.Time) > scaleUp_stabilizationWindow {
+		if hasLargerInstanceType && vsp.Status != nil && time.Since(vsp.Status.TargetStatuses[namespacedIGName].LastTransitionTime.Time) > scaleUp_stabilizationWindow {
 			nextBiggerInstance := instanceTypeRange.InstanceTypes[currInstanceTypeIndex+1]
```

Comment:
Can we create and use a nextSmallerInstance with value`instanceTypeRange.InstanceTypes[currInstanceTypeIndex-1]` as well?